### PR TITLE
MB-46912: Document Python UDFs — REST API

### PR DESCRIPTION
**Pull request by parshimers**
This is just from my explorations, hopefully it isn't too awful. 

* Gerrit change for Python UDFs, including Analytics Library REST API, is at [cbas-core+155852](http://review.couchbase.org/c/cbas-core/+/155852)
* Pull request to add Analytics Library REST API to the nav is at [couchbase/docs-server#2152](https://github.com/couchbase/docs-server/pull/2152)

Docs issue: [MB-46912](https://issues.couchbase.com/browse/MB-46912)

### CODE REVIEW

File: src/analytics-library/extensions/spring/get_collection/http-response.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+*Response 200*
+
+``` json
+[ {
+  "scope" : "travel-sample/inventory",
+  "hash_md5" : "<MD5 hash>",
```
Comment (simon-dew) [inline src/analytics-library/extensions/spring/get_collection/http-response.md:6]: Please check this response and add a real example MD5 hash, if it's not too long

File: src/analytics-library/swagger/analytics-library.yaml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    running the Analytics service.
+
+produces:
+  - application/json
+
+consumes:
+  - multipart/form-data
+
+paths:
+  /analytics/library/:
```
Comment (simon-dew) [inline src/analytics-library/swagger/analytics-library.yaml:24]: Is there really no way to get information about all the libraries and functions in a single scope, or all the functions in a single library? Similar to the Links REST API?

**Review by simon-dew — CHANGES_REQUESTED** [review]
This is great, thanks for contributing this @parshimers. I have added some extra extension markdown to included links, comments, and notes in the generated documentation. I have also added some example CURL requests and responses. Also added a couple of review comments — please can you check these?

File: src/analytics-library/swagger/analytics-library.yaml (hunk 2 above)
Comment (parshimers) [reply src/analytics-library/swagger/analytics-library.yaml:24]: Nope. Not yet anyway. Could definitely add it at some point without too much trouble.

File: src/analytics-library/extensions/spring/get_collection/http-response.md (hunk 1 above)
Comment (parshimers) [reply src/analytics-library/extensions/spring/get_collection/http-response.md:6]: Cool, added.

**Comment by parshimers** [comment]
Nice, thanks for the fixes on the noob mistakes. I added that one hash in the example response.

**Review by simon-dew — APPROVED** [review]
👍 LGTM

**Pull request merged**